Automattic/wp-calypso
# Purchases: Use Redux reducer for SiteSettings

Part of https://github.com/Automattic/wp-calypso/issues/5046

This PR updates components in `my-sites/site-settings` which have a `purchases` prop for the list of purchases associated with the current site. The following 2 components are now connected to the store using connect: 
- `my-sites/site-settings/main.jsx`
- `my-sites/site-settings/delete-site.jsx`

Other components just receive the purchases or fetching status as props.

### Testing instructions
- `git fetch origin && git checkout update/purchases-redux && make run` or use calypso.live
- Go to Site Settings (general section)
- For a site that has no purchase check that you can click on `Delete Site` and be redirected to `/settings/delete-site/:site`
- For a site that has active subscriptions, check that the popup message appears telling you to disable your purchases
- Try deleting a site by going directly to `/settings/delete-site/:site`, it should work for the former but not the latter
- Move around in Settings and check that no component is broken, especially `General`, `Export` and `Guided Transfer` which have been modified

#### Side note
A few components in Site Settings use the now deprecated `valueLink` helper for 2-way binding. This causes some error messages to display in the console:
![image](https://cloud.githubusercontent.com/assets/230230/16729404/76d06c9c-476c-11e6-92f7-076a6027beba.png)
This PR does not fix those

### Reviews

Test live: https://calypso.live/?branch=update/purchases-redux
